output.go

```diff
@@ -5,743 +5,571 @@ import (
+				errorCodeTS := timeseries{
+					name:       "http_error_code",
+					metricType: metrics.Gauge,
+					tags:       ts.tags,
+				}
+				ms.store[errorCodeTS] = value{newValue, 1}
+				log.Tracef("Created %q from %q", errorCodeTS.name, ts.name)
+			}()
+
+			// TODO: We should revisit this. This keeps the old behavior, but I'm not sure having the status code as the
+			// value of a gauge is actually useful.
+			func() {
+				strCode, _ := ts.tags.Get("status")
+				newValue, _ := strconv.ParseFloat(strCode, 32)
+				statusCodeTS := timeseries{
+					name:       "http_status_code",
+					metricType: metrics.Gauge,
+					tags:       ts.tags,
+				}
+				ms.store[statusCodeTS] = value{newValue, 1}
+				log.Tracef("Created %q from %q", statusCodeTS.name, ts.name)
+			}()
+
+			func() {
+				strCode, _ := ts.tags.Get("proto")
+				newValue, _ := strconv.ParseFloat(strings.TrimPrefix(strCode, "HTTP/"), 32)
+				httpVersionTS := timeseries{
+					name:       "http_version",
+					metricType: metrics.Gauge,
+					tags:       ts.tags,
+				}
+				ms.store[httpVersionTS] = value{newValue, 1}
+				log.Tracef("Created %q from %q", httpVersionTS.name, ts.name)
+			}()
+
+		// Rename to http_requests_failed_total
+		case "http_req_failed":
+			newTS := ts
+			newTS.name = "http_requests_failed_total"
+			ms.store[newTS] = v
+			log.Tracef("Created %q from %q", newTS.name, ts.name)
+
+		// Add _bytes suffix to data_sent and data_received.
+		case "data_sent", "data_received":
+			wihtSuffixTS := ts
+			wihtSuffixTS.name = wihtSuffixTS.name + "_bytes"
+			ms.store[wihtSuffixTS] = v
+			log.Tracef("Created %q from %q", wihtSuffixTS.name, ts.name)
+
+		case "http_req_duration":
+			// Tweak name and units.
+			func() {
+				newTS := ts
+				newTS.name = "http_total_duration_seconds"
+				v.value /= 1000 // convert from ms.
+				ms.store[newTS] = v
+				log.Tracef("Created %q from %q", newTS.name, ts.name)
+			}()
+			// Additionally, use the labels of this metric to create a made up "resolve" phase with value of zero.
+			func() {
+				newTS := ts
+				newTS.name = "http_duration_seconds"
+				newTS.tags = newTS.tags.With("phase", "resolve")
+				v.value = 0
+				ms.store[newTS] = v
+				log.Tracef("Created %s{phase=%q} from %q", newTS.name, "resolve", ts.name)
+			}()
+
+		case "iteration_duration":
+			newTS := ts
+			newTS.name = "iteration_duration_seconds"
+			v.value /= 1000 // convert from ms.
+			ms.store[newTS] = v
+			log.Tracef("Created %q from %q", newTS.name, ts.name)
+
+		// Squash multiple duration metrics into one with a "phase" label, which for historical reasons have slightly
+		// different names to k6 phases.
+		// Note that SM also outputs a http_duration_seconds{phase="resolve"} metric, but this one is hardcoded to zero
+		// and generated from http_requ_duration.
+		case "http_req_blocked", "http_req_connecting", "http_req_receiving", "http_req_sending", "http_req_tls_handshaking", "http_req_waiting":
```

Comment:
Old code filtered out the `blocked` and `sending` phases, while the new does not. Thanks @d0ugal for noticing this!